src/plugins/artifact/editor/ContributionList.test.js

```diff
@@ -152,11 +154,47 @@ function createTestData(): * {
 }
 
 describe("ContributionList", () => {
-  it("renders some test data in the default state", () => {
+  // Render a contribution list with the above test data.
+  function render() {
     const data = createTestData();
-    const result = reactTestRenderer.create(
+    const result = mount(
       <ContributionList graph={data.graph()} adapters={data.adapters()} />
     );
-    expect(result).toMatchSnapshot();
+    return result;
+  }
+
+  // Select the unique <option> whose text matches the given patern.
+  function simulateSelect(container: ReactWrapper, pattern: RegExp): void {
+    const targetOption = container
+      .find("option")
+      .filterWhere((x) => pattern.test(x.text()));
+    expect(targetOption).toHaveLength(1);
+    container
+      .find("select")
+      .simulate("change", {target: {value: targetOption.prop("value")}});
+  }
+
+  it("renders some test data in the default state", () => {
+    const result = render();
+    expect(result.find("tbody tr")).toHaveLength(5);
```

Comment:
why 5? could this be rewritten as testNodes.length or such?

Reply:
Could be `createTestData().graph().getAllNodes().length`, although the other case of this assertion would have to be either still `(2)` or something weird that recreates the filter function (i.e., not useful).

Really, though, I might as well just remove these assertions: they're redundant with the snapshot test.

Reply:
> Really, though, I might as well just remove these assertions: they're redundant with the snapshot test.

Although, after removing them, we should keep in mind that if the snapshots undergo a ton of churn, then we could easily replace them with more direct tests like this.